llm_on_ray/inference/predictors/transformer_predictor.py

```diff
@@ -102,33 +112,28 @@ def streaming_generate(self, prompt, streamer, **config):
             **config,
         )
 
-    def generate(
-        self, prompts: Union[str, List[str]], **config
-    ) -> Union[GenerateResult, List[GenerateResult], None]:
+    def generate(self, input: GenerateInput, **config) -> GenerateOutput:
+        if isinstance(input, tuple):
+            raise TypeError("TransformerPredictor doesn't support MLLM input.")
+
+        # Convert prompts to list
+        prompts = [input] if isinstance(input, SinglePromptInput) else input
+
         input_ids, input_length = self.tokenize_inputs(prompts)
         gen_tokens = self.model.generate(
             input_ids, stopping_criteria=self.stopping_criteria, **config
         )
 
         decode_result = self.tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)
-
-        if isinstance(prompts, str):
-            return GenerateResult(
-                text=decode_result,
+        results = [
+            ModelGenerateResult(
+                text=decode_result[i],
                 input_length=input_length,
                 generate_length=gen_tokens.size()[1] - input_length,
             )
-        elif isinstance(prompts, List):
-            return [
-                GenerateResult(
-                    text=decode_result[i],
-                    input_length=input_length,
-                    generate_length=gen_tokens.size()[1] - input_length,
-                )
-                for i in range(len(prompts))
-            ]
-
-        return None
+            for i in range(len(prompts))
+        ]
+        return results[0] if isinstance(prompts, SinglePromptInput) else results
```

Comment:
Should it be ‘input’ here? Because prompts always is a List here.

Reply:
thanks, good catch!